src/weekly/mod.rs

```diff
@@ -88,6 +153,11 @@ impl WeeklyCalendar {
         u64::from(measured)
     }
 
+    /// Return true if the calendar contains no time-windows.
+    pub fn is_empty(&self) -> bool {
+        self.windows.iter().count() == 0
```

Comment:
`self.windows.iter().next().is_some()` is better for general iterators that don't have size hints I'd say.